LibreNMS/Alert/Transport/Snmptrap.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/* Copyright (C) 2015 Daniel Preussker <f0o@devilcode.org>
+ ** This program is free software: you can redistribute it and/or modify
+ ** it under the terms of the GNU General Public License as published by
+ ** the Free Software Foundation, either version 3 of the License, or
+ ** (at your option) any later version.
+ **
+ ** This program is distributed in the hope that it will be useful,
+ ** but WITHOUT ANY WARRANTY; without even the implied warranty of
+ ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ ** GNU General Public License for more details.
+ **
+ ** You should have received a copy of the GNU General Public License
+ ** along with this program.  If not, see <https://www.gnu.org/licenses/>. */
+
+/**
+ ** SNMP Trap Transport
+ **
+ ** @author Cloud Delivery Team <ttc-cloud-delivery@millicom.com>
+ ** @copyright 2022 Cloud Delivery Team, Tigo Technology Center
+ ** @license GPL
+ **/
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Config;
+
+class Snmptrap extends Transport
+{
+    /**
+     * @param  array  $obj
+     * @param  array  $opts
+     */
+    public function deliverAlert($obj, $opts)
+    {
+        $host = $this->config['snmptrap-destination-host'];
+        if (! empty($this->config['snmptrap-destination-port'])) {
+            $port = $this->config['snmptrap-destination-port'];
+        } else {
+            /* Default SNMP trap port */
+            $port = '162';
+        }
+        $transport = $this->config['snmptrap-transport'];
+        $trapdefinition = $this->config['snmptrap-definition'];
+        $pdu = $this->config['snmptrap-pdu'];
+        $community = $this->config['snmptrap-community'];
+        $binary = implode([Config::get('snmptrap')]);
+        $mibdir = $this->config['mib-dir'];
+
+        /**
+         * @param  array  $obj
+         */
+        return $this->contactSnmptrap($binary, $mibdir, $transport, $host, $port, $community, $trapdefinition, $pdu, $obj);
+    }
+
+    /**
+     * Returns if the call was successful
+     *
+     * @return bool
+     */
+    private function contactSnmptrap(string $binary, string $mibdir, string $transport, string $host, string $port, string $community, string $trapdefinition, string $pdu, array $obj)
+    {
+        $binary_opts = '';
+        switch ($pdu) {
+            case 'TRAPv2':
+                 $binary_opts = '-v 2c';
+                break;
+            case 'INFORM':
+                $binary_opts = '-v 2c -Ci';
+                break;
+            default:
+                echo 'This should not happen!!!';
+                break;
+        }
+
+        $msgsingle = preg_replace('~\R~', ' ', $obj['msg']);
+
+        putenv('SNMP_PERSISTENT_FILE=/tmp/snmpapp.conf.$USER');
+
+        //exec('/usr/bin/echo'
+        exec($binary
```

Comment:
This allows cli injection.
Try using Symfony Process class (other examples can be found in source) and send an array to it.
(it also allows you to specify env vars)

Reply:
Looking at Signal.php, is something like this what you expect?:

`    public function contactSignal($obj, $opts)
    {   
        exec($opts['path']
           . ' --dbus-system send'
           . $opts['recipient-type']
           . $opts['recipient']
           . ' -m ' . escapeshellarg($obj['title']));

        return true;
    }
`

Reply:
No, something like this:
```php
use Symfony\Component\Process\Process;
 ...
$process = new Process(
    [Config::get('snmptrap'), '--dbus-system', 'send', $opts['recipient-type'], $opts['recipient'], '-m', $obj['title']],
    null,
    ['SNMP_PERSISTENT_FILE'=>'/tmp/snmpapp.conf.$USER']
);
$process->run();

return $process->isSuccessful();
```